tools/utils.cc

```diff
@@ -111,6 +126,10 @@ int tool_app_template::run_async(int argc, char** argv, noncopyable_function<int
     }
 
     tools::utils::configure_tool_mode(app_cfg, _cfg.logger_name);
+    // i have to set the smp option before starting the reactor
+    if (unsigned shards = parse_shards_option(argc, argv); shards > 0) {
+        app_cfg.smp_opts.smp.set_value(shards);
+    }
```

Comment:
This is too intrusive. We don't want to run the reactor with X shards just because we want to check an sstable with X shards. What if I run this on my laptop with 4 CPUs and I want to examine an sstable from a node with 64 CPUs?

There is a much simpler way: take the schema of the sstable and set a custom murmur sharder on it, configured with the desired shard count. Then set the schema of the sstable to this custom schema:
```c++
auto schema = sst->schema();
auto new_schema = schema_builder(schema).with_sharder(shard_count, sharding_ignore_msb_bits); // need to also expose an option for sharding_ignore_msb_bits, defaulted to 12 (which is the same default scylla config has)
```
Now we have the schema with the correct sharder, but the problem is that the sstable already loaded the sharding info with the default sharder (with `smp::count` shards).
There is no method to set a new schema. I think our best option is to reopen sstables with the new schema:
```c++
auto new_sst = sstable_manager.make_sstable(.../*derive these from existing sstable*/);
co_await sst->load(...);
// now print sharding info
```